plugins/modules/azure_rm_cdnprofile.py

```diff
@@ -177,6 +194,11 @@ def exec_module(self, **kwargs):
 
         if self.state == 'present':
 
+            curr_identity = response["identity"] if response else None
+            update_identity = False
+            if self.identity:
+                update_identity, self.identity = self.update_managed_identity(curr_identity=curr_identity, new_identity=self.identity)
```

Comment:
@nirarg How did you set 'append=False'? The 'allow_identities_append=True' parameter of the function self.update_managed_identity has a default value, so it is added. If you want to override, you need to pass the function when 'allow_identities_append=False'. So you should pass it to the function according to the arguments of 'append'. Thank you!

Reply:
Hi @Fred-sun,
Yes, I used `append=false`. The `allow_identities_append` variable is used for cases where the `append` attribute is not available. This is not the case here, `append` is being used. Therefore, the default value in the function is `allow_identities_append=True`.
I debugged the code and saw that the `identity` is returned as expected from this function and the result is sent to the API. Therefore, I suspect there is something I miss in the API call or in the SDK code.